hackernews.el

```diff
@@ -422,66 +455,59 @@ Objects are decoded as alists and arrays as vectors.")
       (url-insert-file-contents url)
       (hackernews--parse-json))))
 
-(defun hackernews--retrieve-items (feed n ids &optional append)
-  "Retrieve and render at most N new items from FEED.
+(defun hackernews--retrieve-items ()
+  "Retrieve items associated with current buffer."
+  (let* ((items  (hackernews--get :items))
+         (reg    (hackernews--get :register))
+         (nitem  (length items))
+         (offset (car reg))
+         (ids    (cdr reg)))
+    (dotimes-with-progress-reporter (i nitem)
+        (format "Retrieving %d %s..."
+                nitem (hackernews--feed-name (hackernews--get :feed)))
+      (aset items i (hackernews--read-contents
+                     (hackernews--item-url (aref ids (+ offset i))))))))
+
+(defun hackernews--load-stories (feed n &optional append)
```

Comment:
FYI the `append` argument could just as well be called `register`, as it corresponds entirely to the new `:register` property, but I think `append` makes this function a bit clearer and more readable.

Reply:
Yeah, I made sure I read the docstring before reading the implementation and I can confirm that it felt very natural to call it append!